src/NServiceBus.Core/Persistence/PersistenceStartup.cs

```diff
@@ -20,6 +21,8 @@ public void Run(SettingsHolder settings)
             var resultingSupportedStorages = new List<Type>();
             var diagnostics = new Dictionary<string, object>();
 
+            ValidateSagaAndOutboxPersistence(enabledPersistences);
```

Comment:
Nitpicking, but would it be better to indicate what the validation is about? E.g. `ValidateSagaAndOutboxPersistenceIsSame` or something among the lines? That way I don't have to dive into the implementation to understand what the method is supposed to do.